src/ASBNApp.Frontend/Services/ASBNDataService.cs

```diff
@@ -34,119 +35,95 @@ public async Task<EntryRowModelWithID> GetDay(DateTime? date)
+		/// <param name="id">Id of the object to be deleted.</param>
+		/// <returns>Bool, true representing success, false failure.</returns>
+		public async Task<bool> DeleteDay(int? id)
+		{
+			if (id == null) { return false; }
 
-        public async Task<bool> DeleteDay(int? id)
-        {
-            if(id == null) { return false; }
+			HttpResponseMessage response = await _httpClient.DeleteAsync($"/api/odata/Entry({id})");
+			return response.IsSuccessStatusCode;
+		}
 
-            HttpResponseMessage response = await _httpClient.DeleteAsync($"/odata/Entry({id})");
-            return response.IsSuccessStatusCode;
-        }
 
+		/// <summary>
+		/// Get all entries for a given timeframe.
+		/// </summary>
+		/// <param name="startDate">First day in the week.</param>
+		/// <param name="endDate">Last day in the week.</param>
+		/// <returns>List of <see cref="EntryRowModelWithID"/> objects.</returns>
+		public async Task<List<EntryRowModelWithID>> GetWeek(DateTime? startDate, DateTime? endDate)
+		{
+			try
+			{
+				var json = await _httpClient.GetStringAsync($"/api/odata/Entry?$filter=Date ge {startDate?.ToString("yyyy-MM-dd")} and Date le {endDate?.ToString("yyyy-MM-dd")}&$orderBy=Date");
+				var odata = JsonSerializer.Deserialize<ODataBase<EntryRowModelWithID>>(json);
 
+				return odata.value;
+			}
+			catch (Exception ex)
+			{
+				return new List<EntryRowModelWithID>();
+			}
+		}
 
-        /// <summary>
-        /// Handles saving data for an entire week.
-        /// 
-        /// Only saves entries where the <see cref="EntryRowModelWithID.Note"/> != null, 
-        /// using POST / PATCH on the value for Id (if = null use POST, if != null use PATCH).
-        /// </summary>
-        /// <param name="entries">List of <see cref="EntryRowModelWithID"/>s to save.</param>
-        /// <returns>True if success, false if error during saving.</returns>
-        /// <exception cref="Exception">Throw exception if week wasn't saved successfully.</exception>
-        public async Task<bool> SaveWeek(IEnumerable<EntryRowModelWithID> entries)
-        {
-            foreach(var entry in entries)
-            {
-                if(string.IsNullOrWhiteSpace(entry.Note))
-                {
-                    Console.WriteLine($"Nothing to save here, skipping entry for date {entry.Date}.");
-                    continue;
-                }
+		/// <summary>
+		/// Handles saving data for an entire week.
+		/// 
+		/// Only saves entries where the <see cref="EntryRowModelWithID.Note"/> != null, 
+		/// using POST / PATCH on the value for Id (if = null use POST, if != null use PATCH).
+		/// </summary>
+		/// <param name="entries">List of <see cref="EntryRowModelWithID"/>s to save.</param>
+		/// <returns>True if success, false if error during saving.</returns>
+		/// <exception cref="Exception">Throw exception if week wasn't saved successfully.</exception>
+		public async Task<bool> SaveWeek(IEnumerable<EntryRowModelWithID> entries)
```

Comment:
Maybe that should be renamed to `SaveEntries`. Because this method not really cares about weeks, single days or a complete year